cdap-api/src/main/java/co/cask/cdap/api/workflow/WorkflowConfigurer.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.api.workflow;
+
+import java.util.Map;
+
+/**
+ * Configurer for configuring the {@link Workflow}.
+ */
+public interface WorkflowConfigurer {
+
+  /**
+   * Sets the name of the {@link Workflow}
+   *
+   * @param name of the {@link Workflow}
+   */
+  void setName(String name);
+
+  /**
+   * Sets the description of the {@link Workflow}
+   *
+   * @param description of the {@link Workflow}
+   */
+  void setDescription(String description);
+
+  /**
+   * Sets the map of properties that will be available through the {@link WorkflowSpecification#getProperties()}
+   * at runtime
+   *
+   * @param properties the properties to set
+   */
+  void setProperties(Map<String, String> properties);
+
+  /**
+   * Adds a MapReduce program as a next sequential step in the {@link Workflow}. MapReduce program must be
+   * configured when the Application is defined.
+   *
+   * @param mapReduce name of the MapReduce program to be added to the {@link Workflow}
+   *
+   */
+  void addMapReduce(String mapReduce);
```

Comment:
What happens when the mapreduce of given name is not found in application.
Please document the behavior.